test/Sema/enum_raw_representable.swift

```diff
@@ -148,13 +148,13 @@ do {
   // expected-error@-1 {{binary operator '==' cannot be applied to operands of type 'String' and 'E_32431165'}} expected-note@-1 {{partially matching parameter lists: (String, String)}}
 }
 
-func rdar32431165_overloaded() -> Int { 42 }     // expected-note {{'rdar32431165_overloaded()' produces 'Int', not the expected contextual result type 'E_32431165'}}
-func rdar32431165_overloaded() -> String { "A" } // expected-note {{'rdar32431165_overloaded()' produces 'String', not the expected contextual result type 'E_32431165'}}
+func rdar32431165_overloaded() -> Int { 42 }
+func rdar32431165_overloaded() -> String { "A" }
 
 func test_candidate_diagnostic() {
   func test_argument(_: E_32431165) {}
 
-  let _: E_32431165 = rdar32431165_overloaded() // expected-error {{no 'rdar32431165_overloaded' candidates produce the expected contextual result type 'E_32431165'}}
+  let _: E_32431165 = rdar32431165_overloaded() // expected-error {{cannot convert value of type 'String' to specified type 'E_32431165'}} {{23-23=E_32431165(rawValue: }} {{48-48=) ?? <#default value#>}} 
```

Comment:
Yes, I agree too :) 
Just thought that it was no problem to be more specific since the diagnostic bellow produces this message for a similar situation. But now I think that this may be an "ambiguous" too